hack/tools/update_providers/Makefile

```diff
@@ -0,0 +1,42 @@
+# Go compiler
```

Comment:
We should have the targets here be part of the top-level Makefile, and have them be consistent with the way the top-level Makefile calls other tools targets.

Reply:
+1

Reply:
Updated and squashed the commits. Please take a look again!